lib/web_console/middleware.rb

```diff
@@ -33,6 +33,11 @@ def call(env)
           headers["X-Web-Console-Session-Id"] = session.id
           headers["X-Web-Console-Mount-Point"] = mount_point
 
+          # Remove any previously set Content-Length header because we modify
+          # the body. Otherwise the response will be truncated.
+          # Someone will calculate it again, at least the application server.
+          headers.delete("Content-Length")
```

Comment:
Can you add a test for this, so we don't regress again?

Reply:
I would be glad to do that, but unfortunately I don't have any clue how to test it. 😕

I tried adding a wrong Content-Length to the headers defined in `helper_test.rb` or `middleware_test.rb` to reproduce the bug in the first place, but even without this fix, the response in the test always contains the correct Content-Length. Maybe somewhere in the test-stack the Content-Length gets re-set to the correct value, where other application servers don't override the Content-Length if it's already set (e.g. puma)? My rails-knowledge isn't really that good...

Reply:
Maybe we can explicitly set the `Content-Length` after the injection on line `42` ourselves? Then we can test that the content length is extended after the middleware run? That way, we won't depend on someone else to set the proper content length and therefore test our own behavior.

Reply:
The problem is that I cannot reproduce the bug (without my fix) in a test...

If I set `headers["Content-Length"] = 7` (just an arbitrary obviously wrong value) in the middleware instead of deleting it, the response in the test will always have the correct `Content-Length`, not the previously defined value of 7. It seems like something in the test library is setting `Content-Length` to the correct value all the time, which is not the same behavior as puma.

So if I add a test to check if `Content-Length` is set to the correct value, it will be green even without my fix... which kinda defeats the purpose of writing a test.

Personally, I don't think a library or application has to care about setting `Content-Length`, since this is usually the task of a server. That's why I would prefer deleting the header. I would go so far to say that the behavior of Action Pack is wrong, it shouldn't set the `Content-Length` in the first place.

So to summarize: If we delete `Content-Length` or re-calculate it in the middleware, I have no idea how to write a reasonable middleware test. 😅

Reply:
That being said, I only tried to test it using an Integration Test. I don't know how to properly test the middleware with another kind of test.

Reply:
Maybe you can try to write a unit test (`ActiveSupport::TestCase`) with and execute the middleware manually with [rack-test](https://github.com/rack-test/rack-test). You can also move this behavior to the `WebConsole::Injector` and add another unit test to it, instead to the middleware. I'd be okay with the injector, caring about acting on the content length, even if we have to give it more input/output responsibilities.

Reply:
I just added a test without seeing your comment. 😅 But I added the middleware_test which always will be green because of the previously mentioned `IntegrationTest`.

Executing it manually sounds like a good plan. I'll add that (later this day, hopefully)